libraries/wavemap/include/wavemap/data_structure/volumetric/hashed_chunked_wavelet_octree_block.h

```diff
@@ -52,8 +52,12 @@ class HashedChunkedWaveletOctreeBlock {
 
   void setNeedsPruning(bool value = true) { needs_pruning_ = value; }
   bool getNeedsPruning() const { return needs_pruning_; }
+  bool& getNeedsPruning() { return needs_pruning_; }
+
   void setNeedsThresholding(bool value = true) { needs_thresholding_ = value; }
   bool getNeedsThresholding() const { return needs_thresholding_; }
+  bool& getNeedsThresholding() { return needs_thresholding_; }
```

Comment:
What is the reasoning for reference based access to a member variable?

Reply:
This made it easier to provide read/write access to `needs_thresholding_` in a recursive function in another class. But you're right that this is ugly. I'll look for a cleaner way.

Reply:
Depending on where and how often it's needed maybe just a lambda that utilizes the copy set and get methods would work.

Reply:
Fixed